### PR TITLE
No clone

### PR DESCRIPTION
**Pull request by jonay2000**

### CODE REVIEW

File: src/with_alloc.rs [hunk 1]
```diff
@@ -33,14 +36,42 @@ use core::iter::FromIterator;
...
-    buf: Vec<T>,
+    buf: Vec<MaybeUninit<T>>,
     capacity: usize,
     readptr: usize,
     writeptr: usize,
 }
 
+impl<T> Drop for AllocRingBuffer<T> {
+    fn drop(&mut self) {
+        for i in self.drain() {
```
Comment (NULLx76) [inline src/with_alloc.rs:49]: does `self.drain().foreach(drop)` work? If so that might be a tad nicer

Comment (jonay2000) [reply src/with_alloc.rs]: fixed in latest commit

**Comment by NULLx76** [comment]
Just to confirm, this supersedes #47 right?

**Comment by jonay2000** [comment]
this branch branches from #47  and contains all additions it has plus more. Merging this means we can close #47.

**Pull request merged**